### DISCUSSION TITLE
Add some magit actions to context-menu for mouse interactions

### DISCUSSION THREAD

**Discussion by DivineDominion** [magit/magit#4624] [Ideas]
I sometimes discard hunks here and there when experimenting with code, and noticed I used to do this laid-back with mouse in hand. Which didn't work that well in Emacs with magit.

So I wondered how hard it'd be to add relevant actions in magit buffers to the context menu and whipped up this prototype:
https://gist.github.com/DivineDominion/d39110eef36d739cbc811801639d1847

It works surprisingly well already to discard selections, files, and hunks. Same with stage/unstage. 

I am not very experienced with using the context menu, neither with magit's code, so this was all very experimental and can likely be refactored a bit.

Is this material for a PR? I don't know. But before it gets lost, I figured I might add it here as an idea.

Enjoy, and feedback welcome!

**Comment by tarsius** [comment]
I have very recently started experimenting with that too! (#2554 very recently made it back into the first page of issues and its the last issue from 2016, so I better get that done. :grin: ) I should have something ready in a few days. One area where my code differs is that I am trying to avoid having to write a menu variant of each command.

**Reply by DivineDominion** [reply]
@tarsius I've  documented this on my website and in the process looked into the details to understand the menu better -- I updated the Gist accordingly once I found that the menu variants weren't needed and as a benefit also expose the key binding!

**Reply by tarsius** [reply]
I've worked on it all day but unfortunately ran into numerous bugs and edge-cases.

**Reply by DivineDominion** [reply]
@tarsius can you name a few problems? The things I added work rather well for me. (I updated my website in the meantime)

**Comment by linuxer0x42** [comment]
Very cool idea! Having these options in the right-click menu would be very useful in my opinion.

**Comment by tarsius** [comment]
I have just merged my implementation. :grinning: 

See #4645.

**Reply by DivineDominion** [reply]
Sweet! There seems to be no way to "close" discussions here, so I guess we leave this open :)

**Reply by DivineDominion** [reply]
@tarsius I'd love to add context menu actions to quickly delete a branch from the log view, i.e. right-clicking here:
![image](https://user-images.githubusercontent.com/59080/164394404-c8aae4e9-6cad-459d-8fb5-ca9ba0679f5d.png)

`b k RET` at that point does the right thing already. But sometimes my modal input gets in the way (not `evil` but `xah-fly-keys`), so when I'm getting an overview anyway, a right-click might be just as quick.

Do you plan to add context menus to the log view, or are you open to PR's? Your approach to menus seems to differ from my initial humble proposal a bit, so I'm worried I might not get this right, but would be very happy to chime in and try.

**Reply by tarsius** [reply]
Implemented in ab801de53827a232b7806362fb08ca804f27c6d0.
